tic-tac-toe.js

```diff
@@ -1,188 +1,172 @@
-let players = ["player1", "player2"];
-let states = ["open", "player1won", "player2won", "draw"];
+const Marks = {
+  P1: "X",
+  P2: "O",
+  Empty: ".",
+};
+
+const players = {
+  P1: "player1",
+  P2: "player2",
+};
+
+const states = {
+  open: "open",
+  P1: "player1won",
+  P2: "player2won",
+  dran: "draw",
+};
+
+/*const players = ["player1", "player2"];
+const states = ["open", "player1won", "player2won", "draw"];
+const squareStates = ["not set", "player 1", "player 2"];*/
 
-class TicTacToe {
-    constructor(){
+export class TicTacToe {
+  // Saninn's comment: Usually the functions should be defined AFTER the constructor... but ok...
+  resetGame() {
     this.field = [
-        [".", ".", "."],
-        [".", ".", "."],
-        [".", ".", "."]
-        ];
-    this.squareState = ["not set", "player 1", "player 2"];
-    /* I HAVEN'T USED THIS*/
-    this.currentPlayer = players[0];
-    this.currentState = states[0];
-    this.movesArray = [];
-    };
-    
-    resetGame() {
-        this.field = [
-            [".", ".", "."],
-            [".", ".", "."],
-            [".", ".", "."]
-            ];
-        this.currentPlayer = players[0];
-        this.currentState = states[0];
+      [Marks.Empty, Marks.Empty, Marks.Empty],
```

Comment:
Tip: You could also use [Array.fill()](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/fill)

```
this.field = [
    new Array(3).fill(Marks.Empty),
    new Array(3).fill(Marks.Empty),
    new Array(3).fill(Marks.Empty),
];
```
![image](https://user-images.githubusercontent.com/5490201/139555924-2a221d18-9887-4c09-9219-26d8f5f89ff4.png)

Or even
```
const gridSize = 3;
this.field = new Array(gridSize).fill(
            new Array(gridSize).fill(Marks.Empty), 
        );
```
![image](https://user-images.githubusercontent.com/5490201/139556061-6f0c0379-2061-4676-929b-ce36de90558a.png)

Reply:
Done.